{
  "contract": "0xe4eabdca81e31d9acbc4af76b30f532b6ed7f3bf",
  "tool": "mythril",
  "start": 1640177195.6043384,
  "end": 1640177222.85796,
  "duration": 27.253621578216553,
  "analysis": {
    "error": null,
    "issues": [
      {
        "address": 260,
        "code": "ta)\n    payable\n    public\n    {\n        require(msg.sender == Owner);\n        // <yes> <report> UNCHECKED_LL_CALLS\n        adr.call.value(msg.value)(data);\n    }\n}",
        "debug": "calldata_Honey_32 + 4: 0xfffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffc\ncalldata_Honey_0: 0xba21d62a00000000000000000000000000000000000000000000000000000000\ncalldatasize_Honey: 0x4\n",
        "description": "A possible integer overflow exists in the function `Command(address,bytes)`.\nThe addition or multiplication may result in a value higher than the maximum representable integer.",
        "filename": "/data/demo_analysis/source_code/0xe4eabdca81e31d9acbc4af76b30f532b6ed7f3bf.sol",
        "function": "Command(address,bytes)",
        "lineno": 38,
        "title": "Integer Overflow ",
        "type": "Warning"
      },
      {
        "address": 683,
        "code": "}\n    \n    function Command(",
        "debug": "",
        "description": "A possible transaction order independence vulnerability exists in function withdraw(). The value or direction of the call statement is determined from a tainted storage location",
        "filename": "/data/demo_analysis/source_code/0xe4eabdca81e31d9acbc4af76b30f532b6ed7f3bf.sol",
        "function": "withdraw()",
        "lineno": 36,
        "title": "Transaction order dependence",
        "type": "Warning"
      },
      {
        "address": 828,
        "code": "nce);                       ",
        "debug": "",
        "description": "Multiple sends exist in one transaction, try to isolate each external call into its own transaction. As external calls can fail accidentally or deliberately.\nConsecutive calls: \nCall at address: 922\n",
        "filename": "/data/demo_analysis/source_code/0xe4eabdca81e31d9acbc4af76b30f532b6ed7f3bf.sol",
        "function": "GetFreebie()",
        "lineno": 25,
        "title": "Multiple Calls",
        "type": "Information"
      },
      {
        "address": 828,
        "code": "nce);                       ",
        "debug": "",
        "description": "A possible transaction order independence vulnerability exists in function GetFreebie(). The value or direction of the call statement is determined from a tainted storage location",
        "filename": "/data/demo_analysis/source_code/0xe4eabdca81e31d9acbc4af76b30f532b6ed7f3bf.sol",
        "function": "GetFreebie()",
        "lineno": 25,
        "title": "Transaction order dependence",
        "type": "Warning"
      },
      {
        "address": 922,
        "code": "balance);\n        }              ",
        "debug": "SOLVER OUTPUT:\ncallvalue: 0xde0b6b3a7640001\nretval_828: 0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff\ncalldata_Honey_0: 0x6ea3f1cb00000000000000000000000000000000000000000000000000000000\ncalldatasize_Honey: 0x4\n",
        "description": "In the function `GetFreebie()` a non-zero amount of Ether is sent to msg.sender.\nIt seems that this function can be called without restrictions.",
        "filename": "/data/demo_analysis/source_code/0xe4eabdca81e31d9acbc4af76b30f532b6ed7f3bf.sol",
        "function": "GetFreebie()",
        "lineno": 26,
        "title": "Ether send",
        "type": "Warning"
      },
      {
        "address": 1208,
        "debug": "",
        "description": "This contract executes a message call to an address provided as a function argument. Generally, it is not recommended to call user-supplied addresses using Solidity's call() construct. Note that attackers might leverage reentrancy attacks to exploit race conditions or manipulate this contract's state.",
        "filename": "/data/demo_analysis/source_code/0xe4eabdca81e31d9acbc4af76b30f532b6ed7f3bf.sol",
        "function": "Command(address,bytes)",
        "lineno": 46,
        "title": "Message call to external contract",
        "type": "Warning"
      },
      {
        "address": 1208,
        "debug": "",
        "description": "The return value of an external call is not checked. Note that execution continue even if the called contract throws.",
        "filename": "/data/demo_analysis/source_code/0xe4eabdca81e31d9acbc4af76b30f532b6ed7f3bf.sol",
        "function": "Command(address,bytes)",
        "lineno": 46,
        "title": "Unchecked CALL return value",
        "type": "Informational"
      }
    ],
    "success": true
  }
}